src/wallet/spend.cpp

```diff
@@ -1088,6 +1088,7 @@ static util::Result<CreatedTransactionResult> CreateTransactionInternal(
     coin_selection_params.tx_noinputs_size = 10 + GetSizeOfCompactSize(vecSend.size()); // bytes for output count
 
     // vouts to the payees
+    txNew.vout.reserve(txNew.vout.size() + vecSend.size());
```

Comment:
A few suggestions:
- Looks to me like `txNew.vout` is always empty here, so I'm not sure why we'd bother to add `txNew.vout.size()` unless I'm missing something ?
- There's an additional likely insert for change. Since the reserved size here is exact, that could trigger a reallocation up to the next power-of-two. Makes sense to make this `vecSend.size()+1` just in case.
- I think the txouts could be `std::move`d (in both places) instead to save a copy similar to #30094?
- Any reason not `.reserve()` for the `txNew.vin` as well?

Reply:
Hey @theuni, excellent review, thanks!
1) yes, it's always empty, leaving it out is indeed more readable - and if people modify it in the future, they should just pay attention to this
2) nice, haven't noticed it, added the +1 (added you as co-author)
3) To be consistent with `txNew.vin`, I've changed it to `emplace_back` instead, the performance should be comparable - do you agree?
4) wanted to be focused only on the issue discovered during code review, but this is much better, thanks for the comment, pushed!